clients/js/src/compat.ts

```diff
@@ -292,8 +292,11 @@ function isEthers5Signer(upstream: object): upstream is Ethers5Signer {
 }
 
 function isEthers6Signer(upstream: object): upstream is Signer {
-  // XXX: this will not match if installed ethers version is different!
-  return upstream instanceof AbstractSigner;
+  return (
+    upstream instanceof AbstractSigner ||
+    (Reflect.get(upstream, 'signTypedData') &&
+      Reflect.get(upstream, 'signTransaction'))
```

Comment:
How come you're using Reflect.get, not `upstream.signTypedData && upstream.signTransaction`?

Reply:
There is no specific reason.

Reply:
Reflect.get basically ignores type safety and returns `any`